src/silx/io/dictdumplinks/_schemas.py

```diff
@@ -0,0 +1,14 @@
+from ._vds_types import parse_vds_schema_v1
+from ._ext_types import parse_ext_schema_v1
+from ._link_types import ExternalVirtualLink
+from ._link_types import ExternalBinaryLink
+
+
+def parse_schema(target: dict) -> ExternalVirtualLink | ExternalBinaryLink | None:
+    dictdump_schema = target.get("dictdump_schema")
+    if dictdump_schema == "external_binary_link_v1":
+        ext = parse_ext_schema_v1(target)
+        return ExternalBinaryLink(ext.shape, ext.dtype, ext.sources)
+    if dictdump_schema == "external_virtual_link_v1":
+        vds = parse_vds_schema_v1(target)
+        return ExternalVirtualLink(vds.shape, vds.dtype, vds.sources)
```

Comment:
@t20100 @payno  I'm using `TypedDict` and `NamedTuple` but I'm basically re-inventing `pydantic`. Can we add  `pydantic` as a base dependency to silx? It has the following dependencies

```
annotated-types-0.7.0 pydantic-2.11.3 pydantic-core-2.33.1 typing-extensions-4.13.2 typing-inspection-0.4.0
```

Reply:
Would make sense IMO

Reply:
OK for me too:
- it's available in conda-forge, debian, ubuntu
- it's the standard library to validate inputs according to a schema so if you need to do so, just use it.

I've opened issue #4317 to raise this question.